bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/FindReplaceLogic.java

```diff
@@ -497,11 +516,10 @@ private int findIndex(String findString, int startPosition) {
 	public int findAndSelect(int offset, String findString) {
 		if (target instanceof IFindReplaceTargetExtension3)
 			return ((IFindReplaceTargetExtension3) target).findAndSelect(offset, findString,
-					isActive(SearchOptions.FORWARD),
-					isActive(SearchOptions.CASE_SENSITIVE), isWholeWordSearchAvailableAndActive(), isActive(SearchOptions.REGEX));
+					isActive(SearchOptions.FORWARD), isActive(SearchOptions.CASE_SENSITIVE),
+					isWholeWordSearchAvailable(findString), isActive(SearchOptions.REGEX));
```

Comment:
```suggestion
					isWholeWordSearchAvailableAndActive(findString), isActive(SearchOptions.REGEX));
```

Reply:
Addressed in https://github.com/eclipse-platform/eclipse.platform.ui/commit/a0b224321663e63aa99f60da4a2527f993d68f7d